python/cudf/cudf/core/dataframe.py

```diff
@@ -831,6 +833,66 @@ def __sizeof__(self):
         index = self._index.__sizeof__()
         return columns + index
 
+    def _slice(self: T, arg: slice) -> T:
```

Comment:
Does this not get used underneath `Series.iloc[slice(...)]` as well?

Reply:
`Series.iloc` seems to depend on `ColumnBase.__getitem__`, which later calls `Series.slice`.

I noticed the following:
```python
>>> s = cudf.Series([1,2,3,4,5])
>>> s.iloc[4:-1:-1]
Series([], dtype: int64)
```
which is likely the same issue, investigating.

Reply:
It's the same issue in `ColumnBase.slice`. https://github.com/rapidsai/cudf/blob/b8cb8c72241ca2ac5f4d4064659129e42ddf4dbb/python/cudf/cudf/core/column/column.py#L609-L610

Submitted a quick fix. Although I think the best fix here is to consolidate `Series.iloc` and `DataFrame.iloc` to share the common code path under `Frame._slice`.